rancher2/resource_rancher2_role_template.go

```diff
@@ -33,18 +34,25 @@ func resourceRancher2RoleTemplateCreate(d *schema.ResourceData, meta interface{}
 		return err
 	}
 
-	roleTemplate := expandRoleTemplate(d)
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		roleTemplate := expandRoleTemplate(d)
```

Comment:
What's the specific failure scenario you have in mind? Do you feel this function should be updated to actually return an error object in case of failure?